pkg/util/cache/cache.go

```diff
@@ -249,17 +250,24 @@ func (c *lruCache[K, V]) Unpin(key K) {
 	}
 	item := e.Value.(*cacheItem[K, V])
 	item.pinCount.Dec()
-	if c.waitQueue.Len() > 0 {
+
+	log := log.With(zap.Any("UnPinedKey", key))
+	if item.pinCount.Load() == 0 && c.waitQueue.Len() > 0 {
+		log.Info("Unpin item to zero ref, trigger activating waiters")
 		// Notify waiters
 		collector := c.scavenger.Spare(key)
 		for e := c.waitQueue.Front(); e != nil; e = e.Next() {
 			w := e.Value.(*Waiter[K])
 			if ok := collector(w.key); ok {
+				log.Info("Activating waiter", zap.Any("activated_waiter_key", w.key))
 				w.c.Broadcast()
-			} else {
-				break
+				//we try best to activate as many waiters as possible every time
```

Comment:
this previous break may block all potential waiters if the waiter in the front has a very large size